flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/EmbeddedCompletedCheckpointStore.java

```diff
@@ -62,7 +63,7 @@ public EmbeddedCompletedCheckpointStore(
                 maxRetainedCheckpoints,
                 initialCheckpoints,
                 SharedStateRegistry.DEFAULT_FACTORY.create(
-                        Executors.directExecutor(), initialCheckpoints));
+                        Executors.directExecutor(), initialCheckpoints, RestoreMode.DEFAULT));
```

Comment:
If we changed the default restore mode to `CLAIM`, can we also keep this here?
From my understanding, we use `NO_CLAIM` mode for the safety to not discard checkpoint.

And if we just want to make the safety, I think this deserves a description and maybe we need to refer to FLINK-27114.

Reply:
I've added a comment here and in `StandaloneCompletedCheckpointStore` after an offline discussion